lib/bootloader_pvm.pm

```diff
@@ -125,16 +125,16 @@ sub prepare_pvm_installation {
         die "Boot process restarted too many times" if ($boot_attempt > 3);
         return (bootloader_pvm::prepare_pvm_installation $boot_attempt);
     }
-    # On powerVM we have to process startshell in bootloader
-    if (YuiRestClient::is_libyui_rest_api) {
-        YuiRestClient::setup_libyui();
-    }
 
     assert_screen("run-yast-ssh", 300);
 
     if (!is_upgrade && !get_var('KEEP_DISKS')) {
         prepare_disks;
     }
+
+    # On powerVM we have to process startshell in bootloader
+    YuiRestClient::setup_libyui() if YuiRestClient::is_libyui_rest_api();
```

Comment:
Hmm, from the quick look I don't see anything specific to powerVM in `setup_libyui` and actions being done in `set_libyui_backend_vars` should be enough. What I'm trying to say, is that `connect_to_app` call can be done outside of the bootloader test module.

Reply:
Done, I used setup_libyui() for uniformity, but I guess it's not practical.

Reply:
Ok, I still didn't make myself clear enough. Using `setup_libyui` makes sense, what I'm saying is that when we use `extend=` we don't have to introduce any changes to bootloader code at all and `setup_libyui` can be called at any point of time. Same is true for the zVM. This approach allows us to treat powerVM/zVM same way as we do for qemu, except differences which are expressed in `setup_libyui` already.

Reply:
Are you suggesting that I remove it completely from bootloader_pvm and zVM? If I do this, where would you suggest to add the app connection? Doesn't this change the current format of the test suites?

Reply:
Yes, I believe we can remove those calls from bootloader completely and do it once for all. E.g. in the end of `bootloader_start`, like we do for qemu and other backends. As an alternative we also could init the connection at the first place it's used by using singleton pattern. It has small downside that all previous steps will be executed even when REST API doesn't work, but as we are going to use it for each step, it will be eliminated.

Reply:
we still have YUI_REST_API if we want to skip.

Reply:
I am a bit confused. For the machines that I have not set a condition for Yui connection in the bootloader_start, it's because I am taking advantage of the install shell, before the bootloader is finished, for example: pvm -> https://openqa.suse.de/tests/5822681#step/bootloader_start/37 . If I want to move it after, I will have to do something similar to xen which I don't know if will work for pvm, zkvm and s390x . Otherwise edit the module bootloader, not to end with installation console. Did I misinterpret something?

Reply:
The point is that on powerVM, and I guess it's same for zVM, we can do all the necessary steps at any point of time. Not sure about zVM, but for those we also can get ip address after the installation is booted. That's why I suggest that we do not inject those in the bootloader code and handle them as you do for qemu. Or as an alternative, we can then inject it in all bootloaders, both ways will be consistent.

Reply:
You mean get ip via ssh?

Reply:
I believe that we are too strict with the acceptance criteria, from what I see we should keep `setup_libyui` module, as main motivation is not to use `startshell`. Module will keep things transparent and same for all backends.

Reply:
Seems to be working